casper/src/test/scala/coop/rchain/casper/api/CreateBlockAPITest.scala

```diff
@@ -62,208 +57,209 @@ class CreateBlockAPITest extends FlatSpec with Matchers with EitherValues {
+  //      implicit val lastFinalizedStorage       = node.lastFinalizedStorage
+  //      implicit val synchronyConstraintChecker = SynchronyConstraintChecker[Effect]
+  //      implicit val lastFinalizedHeightConstraintChecker =
+  //        LastFinalizedHeightConstraintChecker[Effect]
+  //      val (response1, response2, response3) = (for {
+  //        engine       <- new EngineWithCasper[Task](casper).pure[Task]
+  //        engineCell   <- Cell.mvarCell[Task, Engine[Task]](engine)
+  //        blockApiLock <- Semaphore[Effect](1)
+  //
+  //        result <- testProgram(blockApiLock)(
+  //                   engineCell,
+  //                   synchronyConstraintChecker,
+  //                   lastFinalizedHeightConstraintChecker
+  //                 )
+  //      } yield result).unsafeRunSync
+  //
+  //      response1 shouldBe a[Right[_, String]]
+  //      response2 shouldBe a[Left[_, String]]
+  //      response3 shouldBe a[Left[_, String]]
+  //      response2.left.value shouldBe "Error: There is another propose in progress."
+  //      response3.left.value shouldBe "Error: There is another propose in progress."
+  //
+  //      ().pure[Effect]
+  //    }
+  //  }
+  //
+  //  val syncConstraintThreshold = 1d / 3d
+  //
+  //  it should "not allow proposals without enough new blocks from other validators" in effectTest {
+  //    TestNode
+  //      .networkEff(genesis, networkSize = 5, synchronyConstraintThreshold = syncConstraintThreshold)
+  //      .use {
+  //        case nodes @ n1 +: n2 +: _ +: _ +: _ +: Seq() =>
+  //          import n1.{logEff, metricEff, span, timeEff}
+  //          val engine                        = new EngineWithCasper[Task](n1.casperEff)
+  //          implicit val blockStore           = n1.blockStore
+  //          implicit val lastFinalizedStorage = n1.lastFinalizedStorage
+  //          implicit val synchronyConstraintChecker =
+  //            SynchronyConstraintChecker[Effect](syncConstraintThreshold)
+  //          implicit val lastFinalizedHeightConstraintChecker =
+  //            LastFinalizedHeightConstraintChecker[Effect](Long.MaxValue)
+  //          for {
+  //            deploys      <- (0 until 3).toList.traverse(i => basicDeployData[Task](i))
+  //            engineCell   <- Cell.mvarCell[Task, Engine[Task]](engine)
+  //            blockApiLock <- Semaphore[Effect](1)
+  //
+  //            b1 <- n1.publishBlock(deploys(0))(nodes: _*)
+  //            b2 <- n2.publishBlock(deploys(1))(nodes: _*)
+  //            _  <- n1.syncWith(n2)
+  //
+  //            _        <- n1.casperEff.addDeploy(deploys(2))
+  //            b3Status <- createBlock(blockApiLock)(engineCell)
+  //
+  //            _ = b3Status.left.value should include(
+  //              "Must wait for more blocks from other validators"
+  //            )
+  //          } yield ()
+  //      }
+  //  }
+  //
+  //  it should "allow proposals with enough new blocks from other validators" in effectTest {
```

Comment:
Do we have similar test like this with the Proposer?

Reply:
No, here's issue https://github.com/rchain/rchain/issues/3277#issuecomment-798109618